src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -264,6 +264,10 @@ protected override async Task StartAsyncCore(CancellationToken cancellationToken
         /// </summary>
         public async Task InitializeAsync(CancellationToken cancellationToken = default)
         {
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
Rather than just returning, I think we should be using the existing CancellationToken.ThrowIfCancellationRequested pattern that we're already using in this file during host startup. E.g. currently we call this _after_ InitializeAsync. Perhaps we just need to call it before as well.

Reply:
thanks. Done!